docs/topics/inline-classes.md

```diff
@@ -1,19 +1,29 @@
 [//]: # (title: Inline classes)
 
-> Inline classes are in [Alpha](#alpha-status-of-inline-classes). This feature may change incompatibly in the future and require manual migration. 
-> We will appreciate your feedback in [YouTrack](https://youtrack.jetbrains.com/issues/KT).
+> Inline classes are in [Beta](components-stability.md). It is almost stable, but migration steps may be required in the future. 
+> We'll do our best to minimize any changes you will have to make.
+> We would appreciate your feedback on the inline classes feature in [YouTrack](https://youtrack.jetbrains.com/issue/KT-42434).
 >
 {type="warning"}
 
 Sometimes it is necessary for business logic to create a wrapper around some type. However, it introduces runtime 
 overhead due to additional heap allocations. Moreover, if the wrapped type is primitive, the performance hit is terrible, 
 because primitive types are usually heavily optimized by the runtime, while their wrappers don't get any special treatment. 
 
-To solve such issues, Kotlin introduces a special kind of class called an *inline class*, which is declared by placing an 
-`inline` modifier before the name of the class:
+To solve such issues, Kotlin introduces a special kind of class called an *inline class*. 
+Inline classes are a subset of value-based classes. They don't have identity and hold values only.
+
+To declare an inline class, use `inline` or `value` modifier before the name of the class.
+Kotlin also requires the `@JvmInline` annotation before class declaration along with the `value` modifier for the JVM backend:
 
 ```kotlin
 inline class Password(val value: String)
+
```

Comment:
we should separate these classes into different code blocks. This code block is incorrect because it can't be used as a whole.
I'd also suggest putting first and second class after the sentence about modifiers, then the sentence about @JvmInline, then the class with annotation.

Reply:
Fixed.
I also thought about another sentence for the annotation :)